db/db.go

```diff
@@ -20,8 +22,34 @@ const (
 	DeduplicatorCache
 	// MailserversCache is a list of mail servers provided by users.
 	MailserversCache
+	// TopicHistoryBucket isolated bucket for storing history metadata.
+	TopicHistoryBucket
+	// HistoryRequestBucket isolated bucket for storing list of pending requests.
+	HistoryRequestBucket
 )
 
+// NewMemoryDB returns leveldb with memory backend prefixed with a bucket.
+func NewMemoryDB() (*leveldb.DB, error) {
+	return leveldb.Open(storage.NewMemStorage(), nil)
+}
+
+// NewIsolatedDB returns instance that ensures isolated operations.
+func NewIsolatedDB(db *leveldb.DB, prefix storagePrefix) PrefixedLevelDB {
```

Comment:
maybe instead of "isolated" or "prefixed" we can use something that uses the word "namespace"...
like `NewDBNamespace(db, name)`.

Reply:
i see Prefixed used for similar purposes in another project, and it is kinda consistent with "storagePrefix". not sure why i used Isolated in some cases

but i don't have anything against Namespace

Reply:
yeah, I know it is used every here and there, but I feel like a namespace is a more concise and good name if we want to isolate some data from some other data :)

Reply:
done